src/components/NewCardForm.js

```diff
@@ -4,3 +4,75 @@ import emoji from 'emoji-dictionary';
 import './NewCardForm.css';
 
 const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]
+
+
+
+class NewCardForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: "",
+      emoji: "",
+    };
+
+  }
+
+  resetState = () => {
+    this.setState ({
+      text: '',
+      emoji: '',
+    });
+  }
+
+
+  onFormChange = (event) => {
+    const text = event.target.name;
+    const value = event.target.value;
+
+    const updatedState = {};
+    updatedState[text] = value;
+    this.setState(updatedState);
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    const { text, emoji} = this.state;
+
+    if (text === '' || emoji === '') return;
```

Comment:
Is there a requirement to make text and emoji required? And if text and emoji is blank... right now, your site fails silently. I had no idea why I couldn't create a card for a long time and it's because your site didn't tell me that there was a form validation error!